src_erl/Communication_Layer/http_Nerlserver/src/Client/clientStatem.erl

```diff
@@ -151,9 +151,11 @@ training(cast, {sample,[]}, State = #client_statem_state{msgCounter = Counter})
 
   {next_state, training, State#client_statem_state{msgCounter = Counter+1}};
 
-training(cast, {sample,Vector}, State = #client_statem_state{msgCounter = Counter,workersMap = WorkersMap}) ->
+training(cast, {sample,Body}, State = #client_statem_state{msgCounter = Counter,workersMap = WorkersMap}) ->
   %%    Body:   ClientName#WorkerName#CSVName#BatchNumber#BatchOfSamples
-  [_ClientName,WorkerName,_CSVName, _BatchNumber,BatchOfSamples] = re:split(binary_to_list(Vector), "#", [{return, list}]),
+  {_ClientName, WorkerName, _CSVName, _BatchNumber, BatchOfSamples} = binary_to_term(Body),
```

Comment:
@kapelnik 
I think that it is also support compression, therefore, I think that you should use enveloping functions with names encoder/decoder with compression parameter (for the future).

If you don't want to do it right now then comment % TODO in code 
This comment won't stop this PR